creator-node/src/routes/tracks.js

```diff
@@ -550,26 +551,99 @@ module.exports = function (app) {
       return sendResponse(req, res, errorResponseBadRequest(`Invalid ID: ${encodedId}`))
     }
 
-    const { blockchainId: blockchainIdFromTrack } = await models.Track.findOne({
-      attributes: ['blockchainId'],
-      where: { blockchainId },
-      order: [['clock', 'DESC']]
-    })
-
-    if (!blockchainIdFromTrack) {
-      return sendResponse(req, res, errorResponseBadRequest(`No track found for blockchainId ${blockchainId}`))
-    }
-
-    const { multihash } = await models.File.findOne({
+    let fileRecord = await models.File.findOne({
       attributes: ['multihash'],
       where: {
         type: 'copy320',
-        trackBlockchainId: blockchainIdFromTrack
+        trackBlockchainId: blockchainId
       },
       order: [['clock', 'DESC']]
     })
 
-    if (!multihash) {
+    try {
+      // see if there's a fileRecord in redis so we can short circuit all this logic
```

Comment:
shouldn't we do this only if !fileRecord already?

Reply:
good callout